MAVProxy/modules/mavproxy_link.py

```diff
@@ -749,19 +755,22 @@ def accumulated_statustext(self):
 
     def mavlink_packet(self, msg):
         '''handle an incoming mavlink packet'''
-        type = msg.get_type()
-
-        if type == 'HEARTBEAT' or type == 'HIGH_LATENCY2':
-            sysid = msg.get_srcSystem()
-            if not sysid in self.vehicle_list and msg.type != mavutil.mavlink.MAV_TYPE_GCS:
-                self.vehicle_list.add(sysid)
+        pass
 
     def master_callback(self, m, master):
         '''process mavlink message m on master, sending any messages to recipients'''
-
-        # see if it is handled by a specialised sysid connection
         sysid = m.get_srcSystem()
+        compid = m.get_srcComponent()
```

Comment:
Move this to within the `mtype` block.  I know it's a bit fiddly, but we want to keep this particular part of MAVProxy as fast as possible.